File: 20220222-java-virtual-storage-design-kata/src/test/java/com/katas/DirectoryTest.java
package com.katas;


import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class DirectoryTest {
    @Test
    public void should_add_file() {
        Directory dir1 = new Directory("dir1");
        File file1 = new File("file1", 100);

        dir1.add(file1);

        assertThat(dir1.resources()).containsExactly(file1);
    }

    @Test
    public void should_allow_adding_a_directory() {
        Directory dir1 = new Directory("dir1");
        Directory dir2 = new Directory("dir2");

        dir1.add(dir2);

        assertThat(dir1.resources()).containsExactly(dir2);
    }

    @Test
    public void should_calculate_its_size() {
        Directory dir1 = new Directory("dir1");
        dir1.add(new File("file1", 100));
        dir1.add(new File("file2", 50));

        int size = dir1.getSize();

        assertThat(size).isEqualTo(150);
    }
}

File: 20220221-kotlin-virtual-storage-design-kata/src/test/kotlin/com/katas/DirectoryTest.kt
package com.katas

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test


internal class DirectoryTest {
    @Test
    internal fun `should allow adding a file`() {
        val directory = Directory("dir1")
        val file = File("file1", 100)

        directory.add(file)

        assertThat(directory.getFiles()).containsExactly(file)
    }

    @Test
    internal fun `should allow adding a directory`() {
        val directory1 = Directory("dir1")
        val directory2 = Directory("dir2")

        directory1.add(directory2)

        assertThat(directory1.getFiles()).containsExactly(directory2)
    }

    @Test
    internal fun `should allow deleting a file`() {
        val file1 = File("file1", 100)
        val file2 = File("file2", 200)
        val directory = Directory(name = "dir1", files = setOf(file1, file2))

        directory.delete(file1)

        assertThat(directory.getFiles()).containsExactly(file2)
    }
}